kedro/framework/cli/starters.py

```diff
@@ -586,8 +609,9 @@ def _make_cookiecutter_args(
 def fetch_template_based_on_add_ons(template_path, cookiecutter_args: dict[str, Any]):
     extra_context = cookiecutter_args["extra_context"]
     add_ons = extra_context.get("add_ons")
+    example_pipeline = extra_context.get("example_pipeline")
     starter_path = "git+https://github.com/kedro-org/kedro-starters.git"
-    if add_ons:
+    if example_pipeline:
```

Comment:
This isn't correct. Even if the example pipeline isn't wanted, we might need to fetch from a starter template other than the default. @SajidAlamQB can probably help out here about what the flow should be.

Reply:
@merelcht , thank you. You right, I need to consult with @SajidAlamQB what should we do in case of Spark and Viz and no example option.

Reply:
I think this doesn't need to be changed as we already select the right template with examples in them for pyspark, viz and pyspark & viz. These templates already contain the example pipelines but are removed in `hooks/utils.py` `_handle_starter_setup`. So the current default case is `example=no`. I think we just need a check in `utils.py` to skip `_handle_starter_setup` if `example=yes`.

Reply:
I think for the default template [1-5], options without pyspark and viz, we don't already have example pipelines so we might need to change template to `spaceflights-pandas` in here if `example=yes`.

Reply:
> I think only for the default template [1-5], options without pyspark and viz, we don't already have example pipelines so we might need to change template to `spaceflights-pandas` in here if `example=yes`.

yes, looks like logic should be different for pyspark, viz and without them:
1. without pyspark, viz: If --example==yes, use sfaceflights-pandas starter, elif --example ==no, use default template
2. with pyspark or viz or both: use starter anyway, but we need modifications in  _handle_starter_setup

Is it correct?

Reply:
Yes, this sounds right.